nixos/modules/services/network-filesystems/samba.nix

```diff
@@ -252,6 +182,103 @@ in
         networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ 139 445 ];
         networking.firewall.allowedUDPPorts = mkIf cfg.openFirewall [ 137 138 ];
       })
-    ];
 
+      (lib.mkIf cfg.nmbd.enable {
+        systemd.services.samba-nmbd = {
+          description = "Samba NMB Daemon";
+          documentation = [ "man:nmbd(8)" "man:samba(7)" "man:smb.conf(5)" ];
+
+          after = [
+            "network.target"
+            "network-online.target"
+          ];
+
+          partOf = [ "samba.target" ];
+          wantedBy = [ "samba.target" ];
+          wants = [ "network-online.target" ];
+
+          environment.LD_LIBRARY_PATH = config.system.nssModules.path;
+
+          serviceConfig = {
+            ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
+            ExecStart = "${cfg.package}/sbin/nmbd --foreground --no-process-group ${lib.escapeShellArgs cfg.nmbd.extraArgs}";
+            LimitCORE = "infinity";
+            PIDFile = "/run/samba/nmbd.pid";
+            Slice = "system-samba.slice";
+            Type = "notify";
+          };
+
+          unitConfig.RequiresMountsFor = "/var/lib/samba";
+
+          restartTriggers = [ configFile ];
+        };
+      })
+
+      (lib.mkIf cfg.smbd.enable {
```

Comment:
It seems the options here are now always enabled. `cfg.smbd.enable` defaults to `true`. Shouldn't this include `cfg.enable` (which defaults to false)? E.g. `mkIf (cfg.enable && cfg.smbd.enable)`? @anthonyroussel

Reply:
@skeuchel I was under the impression that the whole block will not be processed if `cfg.enable` on line 186 is false. Did you see smbd being enabled now without enabling the samba module?

Reply:
Thank you for the review, I hadn't thought about it.
I will fix this in another PR.

Reply:
> @skeuchel I was under the impression that the whole block will not be processed if `cfg.enable` on line 186 is false. Did you see smbd being enabled now without enabling the samba module?

Yes, a nightly cross build started failing because it was building `samba` all of a sudden, which apparently fails to cross-compile.

Reply:
https://github.com/NixOS/nixpkgs/pull/340272